### DISCUSSION TITLE
pyprojectx as alternative to install poetry

### DISCUSSION THREAD

**Discussion by houbie** [python-poetry/poetry#5388] [Show and tell]
[Pyprojectx](https://pyprojectx.github.io/) provides a wrapper script that bootstraps the installation of all the development tools used in a project. As such it has the ability to install poetry on demand.

The benefits:
- you can clone and build a project without having to install poetry first
- you can lock the poetry version: this makes your project immune for breaking changes in poetry; as developer you don't need to install/select the required version
- projects become beginner-friendly as nothing needs to be set up

Pyprojectx has a [poetry initializer](https://pyprojectx.github.io/usage/#poetry-project): it installs poetry locally inside the project and starts `poetry init`

[pyprojectx](https://github.com/pyprojectx/pyprojectx) itself shows how you can build a poetry project in a minute:
```
git clone https://github.com/pyprojectx/pyprojectx.git
cd pyprojectx
./pw build
```

IMO it makes sense to add this to the documentation.

**Comment by finswimmer** [comment]
Hello @houbie,

thanks for you suggestion. We already provide instruction for 3 common ways. Adding yet another description would have only little benefit for users.

As you seem to be one of the maintainer of pyprojectx, I would invite you to promote it in the github discussion section. I transfer this issue to it.

fin swimmer

**Comment by DetachHead** [comment]
i personally have been using pyprojectx to install poetry and it's been a far more pleasant experience than using the poetry install script, pipx, or any other global installation method. until i found out about it i regularly ended up with a busted installation due to the flakiness of its setup script (eg. https://github.com/python-poetry/poetry/issues/2449#issuecomment-999182160)

for what it's worth, [pdm's install instructions](https://pdm-project.org/en/latest/#__tabbed_2_6) include pyprojectx, i think it would be good if poetry did as well.
